src/udev/udev-builtin-path_id.c

```diff
@@ -494,9 +494,10 @@ static sd_device *handle_usb(sd_device *parent, char **path) {
         port = strchr(str, '-');
         if (!port)
                 return parent;
+        bus = strndupa_safe(str, port - str);
```

Comment:
I know we have the ALLOCA_MAX stuff now so even though this depends on external input it should be safe, but given it's a format string, maybe you can avoid that at all with something like `"usb-%.*s:%s", str, port - str, port`?